app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java

```diff
@@ -111,20 +112,12 @@ public boolean push(Context ctx,List<Survey> surveys){
             return false;
         }
 
-        //Register for event bus
-        try {
-            register();
-        }catch(Exception e){
-            unregister();
-            register();
-        }
-
         try {
             //Converts app data into sdk events
             postProgress(context.getString(R.string.progress_push_preparing_survey));
             Log.d(TAG, "Preparing survey for pushing...");
 
-            PopulateDB.wipeSDKData();
+            PullDhisSDKDataSource.wipeDataBase();
```

Comment:
@idelcano not is correct from PushController invoked PullShisSDKDataSource. If is necessary remove sdk tables this indicates that the remove is distinct in pull or if remove sdk tables is equal in pull and push should be in other class. I have a doubt, is necessary removed programs, org unit etc in push process?